### ISSUE TITLE
Example script crash

### ISSUE THREAD

**Issue by dflemin3** [ibackus/diskpy#7]
Running the IC_example.py example script gives the following errors:

Traceback (most recent call last):
  File "IC_example.py", line 84, in <module>
    IC.generate()
  File "/astro/users/dflemin3/Desktop/diskpy/diskpy/ICgen/ICgen.py", line 160, in generate
    self.maker.sigma_gen()
  File "/astro/users/dflemin3/Desktop/diskpy/diskpy/ICgen/ICgen.py", line 495, in sigma_gen
    r, sigma = sigma_profile.make_profile(self._parent)
  File "/astro/users/dflemin3/Desktop/diskpy/diskpy/ICgen/sigma_profile.py", line 42, in make_profile
    r, sigma = powerlaw(ICobj.settings, ICobj.T)
  File "/astro/users/dflemin3/Desktop/diskpy/diskpy/ICgen/sigma_profile.py", line 203, in powerlaw
    r = np.array((R/Rd).in_units('1'))
  File "/astro/users/dflemin3/.local/lib/python2.7/site-packages/pynbody/array.py", line 607, in in_units
    **context)
  File "/astro/users/dflemin3/.local/lib/python2.7/site-packages/pynbody/units.py", line 249, in ratio
    raise UnitsException("Not convertible")
pynbody.units.UnitsException: Not convertible

I've been looking through the source and found the following around line 200 in diskpy/ICgen/sigma_profile.py in the powerlaw function:

```
Rd = match_units(pynbody.units.au, Rd)[1]
Mstar = match_units(pynbody.units.Msol, Mstar)[1]
# Molecular weight
m = match_units(m, pynbody.units.m_p)[0]
# Maximum R to calculate sigma at (needed for the exponential cutoff region)
Rmax = rmax*Rd

# Q calculation parameters:
G = SimArray([1.0],'G')
kB = SimArray([1.0],'k')

# Initialize stuff
A = SimArray(1.0,'Msol')/(2*np.pi*np.power(Rd,2))
R = np.linspace(0,Rmax,n_points)
r = np.array((R/Rd).in_units('1'))
```

Since Rd is suppled as SimArray(float,'au'), the (R/Rd).in_units('1') should never work.  When I change R to be SimArray(R,'au'), I get another mysterious error:

/astro/users/dflemin3/Desktop/diskpy/diskpy/disk/_math.py:60: RuntimeWarning: invalid value encountered in divide
  hout = np.sqrt(r_r_r/a)
Traceback (most recent call last):
  File "IC_example.py", line 84, in <module>
    IC.generate()
  File "/astro/users/dflemin3/Desktop/diskpy/diskpy/ICgen/ICgen.py", line 166, in generate
    self.maker.rho_gen()
  File "/astro/users/dflemin3/Desktop/diskpy/diskpy/ICgen/ICgen.py", line 523, in rho_gen
    rho.solve(*_kwargs)
  File "/astro/users/dflemin3/Desktop/diskpy/diskpy/ICgen/rhosolver.py", line 84, in solve
    z, r, rho = calc_rho(self._parent, *_options)
  File "/astro/users/dflemin3/Desktop/diskpy/diskpy/ICgen/rhosolver.py", line 303, in calc_rho
    R = setup_r_bins(IC, r)
  File "/astro/users/dflemin3/Desktop/diskpy/diskpy/ICgen/rhosolver.py", line 153, in setup_r_bins
    h = h_est(r, M, m, T, gamma=1)
  File "/astro/users/dflemin3/Desktop/diskpy/diskpy/disk/_math.py", line 61, in h_est
    return hout.in_units(r.units)
AttributeError: 'numpy.ndarray' object has no attribute 'units'

There's something strange going on with how pynbody is handling units and I can't quite figure it out.  I've encountered this error with my own copy of diskpy and with a fresh clone of the ibackus master repo.   Any ideas on how to resolve this?

**Comment by ibackus** [comment]
Running IC_example.py works for me without throwing any errors.  I think there are two possible issues:
- If you have changed any of the settings, you have to worry about units.  If you did change any settings, please post what you did here.
- Your version of pynbody may be outdated.  Try updating (you should probably do that anyway).  I'm on the master branch, commit a58f77063e42a1b1d596cf4376ff443f8071419c, updated 8/9/2015

**Comment by dflemin3** [comment]
I installed a fresh version of the latest pynbody and I also got a fresh version of diskpy from the ibackus master branch to see if my version has the bug.  When I ran "python IC_example.py", I got the same exact "pynbody.units.UnitsException: Not convertible" error." described above.  Here's the full output for reference:

## [dflemin3@chirimiri ICgen]$ python IC_example.py 

Filenames:

IC_file_name : IC.p
paramName : snapshot.param
directorName : snapshot.director
snapshotName : snapshot.std

---

Sigma profile parameters:

innercut : None
kind : None
outercut : None

---

Rho calculation settings:

r_bin_tol : 0.001
nr : None
zmax : None
nz : 200
min_r_bins : 150

---

Position generator settings:

nParticles : 40411
method : grid

---

Tipsy snapshot generator settings:

mScale : 1.0
metals : 1.0

---

General physical parameters:

Tpower : -0.59
kind : powerlaw
Tmin : 0 K
m : 2.00132 m_p
Tmax : inf K
binsys : None
M : 0.33 Msol
T0 : 332.406 K
eos : isothermal
starMode : single
r0 : 0.5 au
gamma : 1.4

---

ChaNGa run options:

changa_args : 
preset : local
runner_args : 
logfile_name : None

## verbose : True

Sigma profile parameters:

rin : 0.5
kind : powerlaw
outercut : None
power : -1
n_points : 1000
rmax : 2.3
Rd : 1.0 au
Qmin : 1.5
innercut : None
cutlength : 0.3
Settings saved to IC_settings.p

## Initial conditions saved to IC.p

Rho calculation settings:

r_bin_tol : 0.001
nr : None
zmax : None
nz : 200
min_r_bins : 150
Settings saved to IC_settings.p

## Initial conditions saved to IC.p

Filenames:

IC_file_name : IC.p
paramName : snapshot.param
directorName : snapshot.director
snapshotName : snapshot.std

---

Sigma profile parameters:

rin : 0.5
kind : powerlaw
outercut : None
power : -1
n_points : 1000
rmax : 2.3
Rd : 10.0 au
Qmin : 1.0
innercut : None
cutlength : 0.3

---

Rho calculation settings:

r_bin_tol : 0.001
nr : None
zmax : None
nz : 200
min_r_bins : 150

---

Position generator settings:

nParticles : 50000
method : grid

---

Tipsy snapshot generator settings:

mScale : 1.0
metals : 1.0

---

General physical parameters:

Tpower : -0.5
kind : powerlaw
Tmin : 10.0 K
m : 2.0 m_p
Tmax : inf K
binsys : None
M : 1.0 Msol
T0 : 150.0 K
eos : isothermal
starMode : single
r0 : 1.0 au
gamma : 1.4

---

ChaNGa run options:

changa_args : 
preset : local
runner_args : 
logfile_name : None
verbose : True
Settings saved to IC_settings.p
Initial conditions saved to IC.p
Traceback (most recent call last):
  File "IC_example.py", line 84, in <module>
    IC.generate()
  File "/astro/users/dflemin3/.local/lib/python2.7/site-packages/diskpy/diskpy/ICgen/ICgen.py", line 160, in generate
    self.maker.sigma_gen()
  File "/astro/users/dflemin3/.local/lib/python2.7/site-packages/diskpy/diskpy/ICgen/ICgen.py", line 495, in sigma_gen
    r, sigma = sigma_profile.make_profile(self._parent)
  File "/astro/users/dflemin3/.local/lib/python2.7/site-packages/diskpy/diskpy/ICgen/sigma_profile.py", line 42, in make_profile
    r, sigma = powerlaw(ICobj.settings, ICobj.T)
  File "/astro/users/dflemin3/.local/lib/python2.7/site-packages/diskpy/diskpy/ICgen/sigma_profile.py", line 203, in powerlaw
    r = np.array((R/Rd).in_units('1'))
  File "/astro/users/dflemin3/.local/lib/python2.7/site-packages/pynbody/array.py", line 607, in in_units
    **context)
  File "/astro/users/dflemin3/.local/lib/python2.7/site-packages/pynbody/units.py", line 249, in ratio
    raise UnitsException("Not convertible")
pynbody.units.UnitsException: Not convertible

**Comment by dflemin3** [comment]
I think I fixed the issue.  It looks like some numpy functions didn't preserve SimArray units so by making SimArrays more explicit in certain places, it runs fine for me and doesn't change any functionality as far as I can tell.  Here's my changes:

In sigma_profile.py around line 205, I changed

```
    R = np.linspace(0,Rmax,n_points)
    r = np.array((R/Rd).in_units('1'))
```

to

```
    R = SimArray(np.linspace(0,Rmax,n_points),'r_units')
    r = R/Rd
```

where r_units is Rd.units, which should be 'au' and is specified by the user.

Then in rhosolver.py around line 144, I changed

```
    r = np.linspace(0, rmax, nr)
```

to 

```
    r_units = IC.sigma.r_bins.units
    r = SimArray(np.linspace(0, rmax, nr),r_units)
```

With these two simple changed, I get my version of diskpy to run on the newest version of pynbody (I did a git pull yesterday to ensure I was up to date).  I tested it on the example scripts and examined at the snapshots and thankfully they look like they should.

If these changes look good to you, I can do a pull request.
